parcels/kernelaos.py

```diff
@@ -0,0 +1,367 @@
+import inspect
```

Comment:
It may look to your like the same, but using a diff-maanger, you'll see the differences in both files. And yes, of course AOS and SOA still share some functionality and behaviour because of the 'A' (-> Array), but the functions in the basekernel need to work with _all_ structures, not just arrays. This really is the minimal baseline.

Reply:
So, I reviewed the Kernel classes and also considered practical (e.g. the Kernels for lists) and theoretical (e.g. Kernels for trees and the likes) alternatives and reduces the specific code in the specialised Kernel classes to the bare minimum. Therefore, there are 2 new functions: 1 function that updates the loaded fieldsets (as this is evidently unrelated to the particle set structure) and 1 function that does the individual advection of a single particle in SciPy. Now, we cannot get rid of the for-loop overarching that because, for example with lists, that for-loop is replaced by a while-loop in lists that also progresses the node iterator. In trees and lists-of-arrays, this overarching iteration is even more different. That said, I think it does shrink the replicated code by quite a substantial bit. Please have a look on it, see if you agree with that assessment.